src/index.test.tsx

```diff
@@ -4,79 +4,69 @@ import React, { useCallback, useState } from 'react'
 
 import { fireEvent, render } from '@testing-library/react'
 
-import { ServiceProps } from './'
 import {createService, extendService} from '.'
 
 interface User {
   name: string
 }
 
-interface AuthAPI {
-  user?: User
-  logIn: (username: string, password: string) => Promise<User>
-  logOut: () => void
-}
-
 describe('createService', () => {
-  let AuthService: React.FC<ServiceProps<AuthAPI>>
-  let authContext: React.Context<AuthAPI>
-  let useAuthAPI: () => AuthAPI
-  let useAuthService: () => AuthAPI
-  let AuthScreen: React.FC
-  let AuthScreenUsingProps: React.FC<AuthAPI>
-
-  beforeAll(() => {
-    useAuthAPI = () => {
-      const [user, setUser] = useState<User>(undefined)
-
-      const logIn = useCallback(
-        async (username: string, password: string) => {
-          setUser({ name: 'you' })
-          return user
-        },
-        [setUser]
-      )
-
-      const logOut = useCallback(() => {
-        setUser(undefined)
-      }, [setUser])
-
-      return { user, logIn, logOut }
-    }
-
-    [AuthService, useAuthService, authContext] = createService<AuthAPI>(
-      useAuthAPI
+  const useAuthAPI = () => {
+    const [user, setUser] = useState<User | undefined>(undefined)
+
+    const logIn = useCallback(
+      async (username: string, _password: string) => {
+        setUser({ name: username })
```

Comment:
Updated the example service to actually use the provided `username` for added realism.